src/lib/operators/table_scan/column_between_table_scan_impl.cpp

```diff
@@ -132,4 +139,51 @@ void ColumnBetweenTableScanImpl::_scan_dictionary_segment(const BaseDictionarySe
   });
 }
 
+void ColumnBetweenTableScanImpl::_scan_sorted_segment(const BaseSegment& segment, const ChunkID chunk_id,
+                                                      PosList& matches,
+                                                      const std::shared_ptr<const PosList>& position_filter,
+                                                      const OrderByMode order_by_mode) const {
+  resolve_data_and_segment_type(segment, [&](const auto type, const auto& typed_segment) {
+    using ColumnDataType = typename decltype(type)::type;
+
+    if constexpr (std::is_same_v<std::decay_t<decltype(typed_segment)>, ReferenceSegment>) {
+      Fail("Expected ReferenceSegments to be handled before calling this method");
+    } else {
+      auto segment_iterable = create_iterable_from_segment(typed_segment);
+      segment_iterable.with_iterators(position_filter, [&](auto segment_begin, auto segment_end) {
+        auto sorted_segment_search =
+            SortedSegmentSearch(segment_begin, segment_end, order_by_mode, _column_is_nullable, predicate_condition,
+                                boost::get<ColumnDataType>(left_value), boost::get<ColumnDataType>(right_value));
+
+        sorted_segment_search.scan_sorted_segment([&](auto begin, auto end) {
```

Comment:
Yes we can and I did it in commit `906a7625`.

Just for your information: `column_between_table_scan_impl.cpp` and `column_vs_value_table_scan_impl.cpp` could be merged even further (see `_scan_non_reference_segment` for example. I would not do this in this PR though because its not the context of this PR. I would rather focus on seminar related topics)